src/net.cpp

```diff
@@ -910,43 +895,8 @@ static void EraseLastKElements(std::vector<T> &elements, Comparator comparator,
     elements.erase(elements.end() - eraseSize, elements.end());
 }
 
-/** Try to find a connection to evict when the node is full.
- *  Extreme care must be taken to avoid opening the node to attacker
- *   triggered network partitioning.
- *  The strategy used here is to protect a small number of peers
- *   for each of several distinct characteristics which are difficult
- *   to forge.  In order to partition a node the attacker must be
- *   simultaneously better at all of them than honest peers.
- */
-bool CConnman::AttemptToEvictConnection()
+[[nodiscard]] Optional<NodeId> SelectNodeToEvict(std::vector<NodeEvictionCandidate>&& vEvictionCandidates)
```

Comment:
fbba7d8aada5b1d7a63ad4133 add `#include <optional.h>` header

Reply:
Now including `optional.h`.

Regarding the naming: I see your point but given `NodeEvictionCandidate` inputs I think `SelectNodeToEvict` works.